doc/developers/contributing.rst

```diff
@@ -207,6 +207,104 @@ and Cython optimizations.
    <http://astropy.readthedocs.org/en/latest/development/workflow/development_workflow.html>`_
    sections.
 
+Rebasing and squashing commits
+------------------------------
+
+Resolving merge conflicts and Rebasing
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you changed the same part of the same file differently in the two branches you’re merging
+together, Git won’t be able to merge them cleanly. This is called a merge conflict. To resolve a
+merge conflict you can either:
+	
+	* Open the files manually and resolve them as git adds standard resolution markers to the
+	  files that have conflicts.
+
+	* Use a graphical tool by running ``git mergetool``. This tool will walk you through all the
+	  conflicts.
+
+.. note::
+	Detailed guide on solving merge conflicts can be found on `git-scm.com
+	<http://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging#Basic-Merge-Conflicts>`_
+
+You should generally use ``rebase`` over ``merge`` to integrate changes from one branch into
+another branch. Merging creates a "merge commit" every time you incorporate upstream changes.
+This pollutes the feature branch's history and makes it hard for other developers to understand
+the history of the project. Whereas ``rebase`` effectively rewrites the project history by creating
+brand new commits for each commit in the original branch. This creates a much cleaner project history.
+To rebase:
+
+	1. Switch to feature branch::
+
+		$ git checkout my-feature-branch
+
+	2. Make a backup just in case::
+
+		$ git branch my-feature-branch-bak my-feature-branch
```

Comment:
Do we need this? There is reflog in extreme cases no? It would be better if we have it simple I feel